Assets/RTLTMPro/Scripts/Editor/ConvertToTextMeshPro.cs

```diff
@@ -0,0 +1,234 @@
+                TextAnchor.LowerLeft    => TextAlignmentOptions.BottomLeft,
+                TextAnchor.LowerCenter  => TextAlignmentOptions.Bottom,
+                TextAnchor.LowerRight   => TextAlignmentOptions.BottomRight,
+                _                       => throw new ArgumentOutOfRangeException()
+            };
+
+            var possibleFonts = AssetDatabase.FindAssets($"t:{nameof(TMP_FontAsset)}")
+                .Select(AssetDatabase.GUIDToAssetPath)
+                .Select(AssetDatabase.LoadAssetAtPath<TMP_FontAsset>);
+
+            bool IsMatchingFont(TMP_FontAsset f)
+            {
+                //We need to use reflection because the font property is null when population mode is set to static
+                const BindingFlags flags = BindingFlags.Instance | BindingFlags.GetField | BindingFlags.Public | BindingFlags.NonPublic;
+                var fontSourceFile = typeof(TMP_FontAsset)
+                    .GetField("m_SourceFontFile_EditorRef", flags)
+                    !.GetValue(f) as Font;
+
+                return fontSourceFile == font;
+            }
+
+            to.font     = possibleFonts.FirstOrDefault(IsMatchingFont);
+            to.fontSize = fontSize;
+
+            to.fontStyle = fontStyle switch
+            {
+                FontStyle.Normal        => FontStyles.Normal,
+                FontStyle.Bold          => FontStyles.Bold,
+                FontStyle.Italic        => FontStyles.Italic,
+                FontStyle.BoldAndItalic => FontStyles.Bold | FontStyles.Italic,
+                _                       => throw new ArgumentOutOfRangeException()
+            };
+
+            to.lineSpacing      = lineSpacing;
+            to.richText         = supportRichText;
+            to.fontSizeMax      = resizeTextMaxSize;
+            to.fontSizeMin      = resizeTextMinSize;
+            to.enableAutoSizing = resizeTextForBestFit;
+            to.enabled          = enabled;
+            to.maskable         = maskable;
+            to.material         = material;
+            to.raycastTarget    = raycastTarget;
+            to.raycastPadding   = raycastPadding;
+
+            if (EditorGUI.EndChangeCheck())
+                Undo.RecordObject(to, "Copy from old text");
+        }
+
+        [MenuItem("CONTEXT/TextMeshProUGUI/Convert to RTLTMPro", true)]
+        public static bool ConvertTextMeshProUGUIValidate(MenuCommand command) 
+            => command.context is TextMeshProUGUI and not RTLTextMeshPro;
+
+        [MenuItem("CONTEXT/TextMeshProUGUI/Convert to RTLTMPro")]
+        public static void ConvertTextMeshProUGUI(MenuCommand command)
+        {
+            var text = (TextMeshProUGUI)command.context;
+            Convert(text);
+        }
+        
+        /// <summary>
+        /// Unfortunately I'm not sure how to do GUID swapping, so references to the text component will break, but at least it should make the transition much safer and quicker
```

Comment:
I thought this asset handled the conversion to TMPro without losing references. Breaking existing references is not an option, there are thousands.

Reply:
You do raise a valid point though, if we go from TMP -> RTLTMP we would still potentially lose some references.

Reply:
I meant references in other components, #2004 only handled references in code, but not from other non-code components.